### PR TITLE
Script to manage annotations

### PR DESCRIPTION
**Pull request by bfhealy**
This PR adds a new tool (scope_manage_annotation.py) that manages Fritz annotations. The tool takes either one source or a CSV file of many, and for each one posts, revises, or deletes a specified annotation. The script performs several checks on the inputs and communicates the status of each run.

### CODE REVIEW

File: tools/scope_manage_annotation.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+import argparse
+import pandas as pd
+from scope.fritz import api
+
+
+def manage_annotation(action, source, group_ids, token, origin, key, value):
+
+    # Forgiving definitions of actions
```
Comment (mcoughlin) [inline tools/scope_manage_annotation.py:9]: NICE!

File: tools/scope_manage_annotation.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    if origin is None:
+        print('Error: please specify origin to %s' % action)
+        return
+
+    if key is None:
+        print('Error: please specify key to %s' % action)
+        return
+
+    # check if source is single object or csv file of many
+    if source[-4:] == '.csv':
```
Comment (mcoughlin) [inline tools/scope_manage_annotation.py:27]: How about using .endswith?

File: tools/scope_manage_annotation.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        file = pd.read_csv(source)  # modify input formats to prepare for loop
+        obj_ids = file['obj_id']
```
Comment (mcoughlin) [inline tools/scope_manage_annotation.py:29]: Maybe gracefully throw error if this header missing?

File: tools/scope_manage_annotation.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                annot_value = [x for x in annot_data.values()][0]
+
+                # if match is found, perform action
+                if (key == annot_name) & (origin == annot_origin):
+                    matches += 1
+                    if action == 'update':
+                        value = values[i]
+
+                        # Check value if performing update or post actions
+                        if value is None:
```
Comment (mcoughlin) [inline tools/scope_manage_annotation.py:66]: Maybe raise an error here?

File: tools/scope_manage_annotation.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+                    % (origin, key, obj_id)
+                )
+
+        # if posting new annotation, skip search for exisiting ones
+        elif action == 'post':
+            value = values[i]
+
+            # Check value if performing update or post actions
+            if value is None:
+                print('Error: please specify annotation value to update or post.')
```
Comment (mcoughlin) [inline tools/scope_manage_annotation.py:127]: Raise an error here too.

**Review by mcoughlin — CHANGES_REQUESTED** [review]
@bfhealy nice! a few minor comments, then can get this in.

**Comment by bfhealy** [comment]
@mcoughlin Thanks, just pushed your requested changes!

**Pull request merged**